pkg/controller/lifecycle/actuator.go

```diff
@@ -110,6 +120,24 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, ex *extensionsv
 		if internalShootConfig.EgressFilter != nil {
 			blackholingEnabled = internalShootConfig.EgressFilter.BlackholingEnabled
 			staticFilterList = internalShootConfig.EgressFilter.StaticFilterList
+			if internalShootConfig.EgressFilter.WorkerSpecific != nil {
+				workerGroupBlackholingEnabled = make(map[string]bool)
+				for _, worker := range cluster.Shoot.Spec.Provider.Workers {
+					shootWG := worker.Name
+					found := false
+					for _, configWG := range internalShootConfig.EgressFilter.WorkerSpecific.Groups {
+						if strings.EqualFold(shootWG, configWG) {
+							found = true
+							break
+						}
+					}
+					if found {
+						workerGroupBlackholingEnabled[shootWG] = internalShootConfig.EgressFilter.WorkerSpecific.BlackholingEnabled
+					} else {
+						workerGroupBlackholingEnabled[shootWG] = blackholingEnabled
+					}
```

Comment:
fancy some generics, do we? ;)
I was boycotting them originally, because Go didn't have them until 1.18. I felt the Go authors didn't add them on purpose to keep the language simple and lean. The decision to add them eventually felt kind of coerced to me. 

I'll leave it up to you. If you want generics, you get generics np.

Reply:
added generics to avoid the nested loop